### PR TITLE
Some table fixing

### LINKED ISSUE #14943: Clicking on a table will open the table crafting menu.
You're supposed to do this by dragging the table onto yourself.

Also, dragging tables onto yourself will not give the little box icon under your mouse to indicate it is a drag-able thing.

Also it's kind of inconsistent feeling that there HAS to be an item on a table in order to bring up table crafting. I think any time you drag a table onto yourself, item or not, it should bring up the menu.

### PR DESCRIPTION
**Pull request by lzimann**
Fixes #14943

### CODE REVIEW

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -95,7 +95,7 @@
 	if(tableclimber && tableclimber != user)
 		tableclimber.Weaken(2)
 		tableclimber.visible_message("<span class='warning'>[tableclimber.name] has been knocked off the table", "You're knocked off the table!", "You see [tableclimber.name] get knocked off the table</span>")
-	..()
+	return
```
Comment (neersighted) [inline code/game/objects/structures/tables_racks.dm]: No need for the implied return at the end.

File: code/modules/crafting/table.dm [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
 	var/list/table_contents = list()
 	var/viewing_category = 1 //typical powergamer starting on the Weapons tab
 	var/list/categories = list(CAT_WEAPON,CAT_AMMO,CAT_ROBOT,CAT_FOOD,CAT_MISC)
 
 
-/obj/structure/table/MouseDrop(atom/over)
-	if(over != usr || !usr.IsAdvancedToolUser())
+/obj/structure/table/MouseDrop(mob/living/user)
+	if(!usr.IsAdvancedToolUser())
```
Comment (neersighted) [inline code/modules/crafting/table.dm]: `user` not `usr`

File: code/modules/crafting/table.dm [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 /obj/structure/table
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
 	var/list/table_contents = list()
 	var/viewing_category = 1 //typical powergamer starting on the Weapons tab
 	var/list/categories = list(CAT_WEAPON,CAT_AMMO,CAT_ROBOT,CAT_FOOD,CAT_MISC)
 
 
-/obj/structure/table/MouseDrop(atom/over)
-	if(over != usr || !usr.IsAdvancedToolUser())
+/obj/structure/table/MouseDrop(mob/living/user)
```
Comment (phil235) [inline code/modules/crafting/table.dm]: Are you sure that ghosts and camera mobs can't call MouseDrop() ?

Comment (bgobandit) [inline code/modules/crafting/table.dm]: even in death mekhi goes on the table

Comment (neersighted) [inline code/modules/crafting/table.dm]: Fair enough

**Pull request merged**